economic_dybdahl_rest/usecases/get_order_lines/get_order_lines.py

```diff
@@ -0,0 +1,173 @@
+
+        draft_orders_json = draft_orders_response.json()
+        pagination = draft_orders_json['pagination']
+        draft_orders = draft_orders_json['collection']
+
+        find_and_add_order_numbers(draft_orders, draft_orders_numbers)
+
+        if len(draft_orders_numbers) > pagination['results']:
+            break
+
+        try:
+            next_page = pagination['nextPage']
+        except KeyError:
+            break
+
+    return draft_orders_numbers
+
+
+def find_and_add_order_numbers(draft_orders, order_numbers_list):
+    for draft_order in draft_orders:
+        order_number = draft_order['orderNumber']
+        order_numbers_list.append(order_number)
+
+
+def get_all_draft_orders_request_and_model_lines(draft_orders_numbers):
+    get_draft_order_api = GetDraftOrderAPI()
+    model_draft_order_lines = []
+    request_draft_order_lines = []
+
+    for draft_order_number in draft_orders_numbers:
+        response = get_draft_order_api.get(draft_order_number)
+        check_status_is_succeeded(response)
+        draft_order_json = response.json()
+        for line in draft_order_json['lines']:
+            request_draft_order_lines.append(line)
+            model_draft_order_lines.append(DraftOrderLine(
+                order_number=draft_order_number,
+                economic_number=line['product']['productNumber'],
+                amount=line['quantity'],
+                available=None
+            ))
+    return request_draft_order_lines, model_draft_order_lines
+
+
+def get_all_product_numbers_from_lines(lines):
+    product_numbers = []
+    for line in lines:
+        product_number = line['product']['productNumber']
+        product_numbers.append(product_number)
+    return product_numbers
+
+
+def get_all_products_to_list(product_numbers):
+    products = []
+    product_numbers_no_duplicates = set(product_numbers)
+    amount_left = len(product_numbers_no_duplicates)
+    while amount_left > 0:
+        get_products_api = GetProducts()
+        products_to_get = []
+        for count, product_number in enumerate(product_numbers_no_duplicates):
+            if count > 19:
+                break
+            products_to_get.append(product_number)
+        product_numbers_no_duplicates = [item for item in product_numbers_no_duplicates if item not in products_to_get]
+
+        response = get_products_api.get(products_to_get)
+        check_status_is_succeeded(response)
+        json_response = response.json()
+
+        products.extend(json_response['collection'])
+
+        amount_left = len(product_numbers_no_duplicates)
+
+    return products
+
+
+def find_and_map_available_to_lines(model_draft_order_lines, products):
+    for draft_order_line in model_draft_order_lines:
+        for product in products:
+            if draft_order_line.economic_number == product['productNumber']:
+                draft_order_line.available = product['inventory']['available']
```

Comment:
Ved godt at det nok ikke er dette kald der "vælter budgettet", men det ville være hurtigere at mappe produkterne i `get_all_products_to_list` således:

```python
{
     "hash_value": { ..details.. }
}
```

For så kan du her bare

```python
for draft_order_line in model_draft_order_lines:
    product = products.get(draft_order_line.economic_number, None)
    if product is not None:
        draft_order_line.available = product['inventory']['available']
```